source/dynamical_systems/src/DynamicalSystem.cpp

```diff
@@ -31,15 +32,30 @@ bool DynamicalSystem<state_representation::CartesianState>::is_compatible(const
       && state.get_reference_frame() != this->get_base_frame().get_reference_frame());
 }
 
+template<>
+bool DynamicalSystem<state_representation::JointState>::is_compatible(const state_representation::JointState& state) {
+  return (state.get_name() == this->get_base_frame().get_name()
+      && state.get_names() == this->get_base_frame().get_names());
+}
+
 template<class S>
 S DynamicalSystem<S>::evaluate(const S& state) const {
   return this->compute_dynamics(state);
 }
 
-template JointState DynamicalSystem<JointState>::evaluate(const JointState& state) const;
+template<>
+JointState DynamicalSystem<JointState>::evaluate(const JointState& state) const {
+  if (this->get_base_frame().is_empty()) {
+    throw exceptions::EmptyBaseFrameException("The base frame of the dynamical system is empty.");
+  }
```

Comment:
I would add the compatibility check here, similar to CartesianState evaluate. The difference is that the Cartesian one has some intermediate logic so can't call `is_compatible` directly, but we can do it here.

```suggestion
  }
  if (!this->is_compatible(is_compatible)) {
    state_representation::exceptions::IncompatibleReferenceFramesException(...);
  }
```

Reply:
Yes that makes sense, more consistency thanks!